src/index.ts

```diff
@@ -134,6 +140,21 @@ export class Connection {
     }
   }
 
+  async transaction<T>(fn: (tx: Transaction) => Promise<T>): Promise<T> {
+    const tx = new Connection(this.config)
+
+    try {
+      await tx.execute('BEGIN')
+      const res = await fn(tx)
+      await tx.execute('COMMIT')
+
+      return res
+    } catch (err) {
+      await tx.execute('ROLLBACK')
```

Comment:
Would errors that come back from a bad query trigger an exception to be raised?

I guess I'm asking, are there other cases we can handle that we'd expect to automatically ROLLBACK?

Reply:
Yeah, they do. Anything that's a bad query will trigger a rollback. We tested this in a script and MySQL automatically rolls back as well for bad queries within a transaction.